wdl/AnnotateTranscriptomeWithGuide.wdl

```diff
@@ -0,0 +1,107 @@
+version 1.0
+
+import "tasks/PBUtils.wdl" as PB
+import "tasks/Utils.wdl" as Utils
+import "tasks/StringTie2.wdl"
+import "tasks/Finalize.wdl" as FF
+
+workflow AnnotateTranscriptomeWithGuide {
+    input {
+        File aligned_bam
+        File aligned_bai
+        File gtf
+        File ref_map_file
+
+        String participant_name
+
+        String gcs_out_root_dir
+    }
+
+    parameter_meta {
+        aligned_bam:        "GCS path to aligned BAM file"
+        aligned_bai:        "GCS path to aligned BAM file index"
+
+        gtf:                "guide GTF (e.g. Gencode38)"
+        ref_map_file:       "table indicating reference sequence and auxillary file locations"
+        participant_name:   "name of the participant from whom these samples were obtained"
+
+        gcs_out_root_dir:   "GCS bucket to store the reads, variants, and metrics files"
+    }
+
+    Map[String, String] ref_map = read_map(ref_map_file)
+
+    String outdir = sub(gcs_out_root_dir, "/$", "") + "/AnnotateTranscriptomeWithGuide/~{participant_name}"
+
+    call StringTie2.Quantify as QuantifyWithoutRetainedIntrons {
+        input:
+            aligned_bam = aligned_bam,
+            aligned_bai = aligned_bai,
+            gtf = gtf,
+            keep_retained_introns = false,
+            prefix = participant_name + ".without_retained_introns"
+    }
+
+    call StringTie2.Quantify as QuantifyWithRetainedIntrons {
+        input:
+            aligned_bam = aligned_bam,
+            aligned_bai = aligned_bai,
+            gtf = gtf,
+            keep_retained_introns = true,
+            prefix = participant_name + ".with_retained_introns"
+    }
+
+    call StringTie2.ExtractTranscriptSequences as ExtractTranscriptsWithoutRetainedIntrons {
+        input:
+            ref_fasta = ref_map['fasta'],
+            ref_fasta_fai = ref_map['fai'],
+            gtf = QuantifyWithoutRetainedIntrons.st_gtf
+    }
+
+    call StringTie2.ExtractTranscriptSequences as ExtractTranscriptsWithRetainedIntrons {
+        input:
+            ref_fasta = ref_map['fasta'],
+            ref_fasta_fai = ref_map['fai'],
+            gtf = QuantifyWithRetainedIntrons.st_gtf
+    }
+
```

Comment:
Can you add in a step to create a sequence dictionary here as well with Picard / GATK `CreateSequenceDictionary`?  Most of the time having the dictionary around is helpful.

Reply:
Will do - I imagine CreateSequenceDictionary or `samtools dict` makes no difference to you?

Reply:
Yup!  Either should work just fine.

Reply:
Done!